packages/eslint/alias-plugins.js

```diff
@@ -0,0 +1,41 @@
+const moduleAlias = require('module-alias');
+const pnpApi = process.versions.pnp ? require('pnpapi') : null; // eslint-disable-line import/no-unresolved
+
+function toFullName(pluginName) {
+  const ESLINT_PREFIX = 'eslint-plugin-';
+  const ORGANIZATION_EXPRESSION = /^(@[\d.A-z-]+)\/(.+)$/;
+  const nameIsFull = pluginName.indexOf(ESLINT_PREFIX) === 0;
+  const nameIsOrganization = ORGANIZATION_EXPRESSION.test(pluginName);
+
+  if (nameIsOrganization) {
+    const [, organizationName, name] = pluginName.match(
+      ORGANIZATION_EXPRESSION,
+    );
+
+    return `${organizationName}/${toFullName(name)}`;
+  }
+
+  return nameIsFull ? pluginName : `${ESLINT_PREFIX}${pluginName}`;
+}
+
+function aliasModuleFrom(relativeFilename = __filename) {
+  return function aliasPlugin(pluginName) {
+    let resolvedPluginPath;
+
+    if (pnpApi) {
+      resolvedPluginPath = pnpApi.resolveRequest(pluginName, relativeFilename);
+    } else {
+      resolvedPluginPath = require.resolve(pluginName, {
+        paths: [relativeFilename],
+      });
+    }
```

Comment:
You should just use `require.resolve` you almost never need to use the pnpapi directly, this is one of those cases

Reply:
I know. But unfortunately `__filename` gives a virtual File System path in **Yarn 2** which can't work correctly in `paths` options of `require.resolve`. Module will be not found with an exception.

Reply:
Do you have a reproduction for that? We have tests making sure it works. I could see that happening if you have `enableGlobalCache: true` but that will be fixed in https://github.com/yarnpkg/berry/pull/2068

Reply:
In any place of your code in Yarn 2 environment try this `console.log(require.resolve('any-of-your-dependency', { paths: [__filename] }))`. It will not find your module even if it is relative from the current file. The same code works without any problems in regular NPM environment

Reply:
It will, and it does. As I said we have tests that cover this, but just for fun I did try exactly what you said and it works, if you have a reproduction I'd love to take a look at it


Some tests that check this:
https://github.com/yarnpkg/berry/blob/f8344493b5cf3948e71211697b50a268b756d8fc/packages/acceptance-tests/pkg-tests-specs/sources/pnp.test.js#L1714
https://github.com/yarnpkg/berry/blob/f8344493b5cf3948e71211697b50a268b756d8fc/packages/acceptance-tests/pkg-tests-specs/sources/pnp.test.js#L609-L612
https://github.com/yarnpkg/berry/blob/f8344493b5cf3948e71211697b50a268b756d8fc/packages/acceptance-tests/pkg-tests-specs/sources/pnp.test.js#L647-L651

Reply:
I realized that mentioned test case works in the Yarn 2 project itself. But in the **linked** module it fails with not found dependency. As you understand I use linked packages for local environments. Such modules are defined as
```json
{
 "devDependencies": {
   "@neutrinojs/eslint": "portal:C:/Projects/Projects_GitHub/neutrino-dev-fork/packages/eslint",
  }
}
```
`yarn install` is called to setup all links in `.yarn/cache` and `.pnp.js`
And after that I can run local script `"lint": "eslint ./src --ext .js,.jsx",` to run the code and test the case

Reply:
Yeah, that makes more sense. I'm fairly certain the issue you're running into was fixed in https://github.com/yarnpkg/berry/pull/2068, could you try with `yarn set version from sources` or by downloading the `bundle` here https://github.com/yarnpkg/berry/actions/runs/346381133?

Reply:
Before: Yarn version `v2.2.2`
After: Yarn version `2.3.3-git.20201112.f5ca1c57`
Result is the same in the linked package: works when `require.resolve('babel-eslint')` and fails  when `require.resolve('babel-eslint', { paths: [__filename] })` 
```
Oops! Something went wrong! :(

ESLint: 7.9.0

Error: Cannot read config file: C:\Users\const\Desktop\yarn-test\.eslintrc.js
Error: Cannot find module 'babel-eslint'
```

Reply:
Could you try using `__dirname` instead?
Again, if you could provide a reproduction I'll be able to look into it

Reply:
**Update:**
After removing `.yarn/cache` and reinstalling dependencies `require.resolve('babel-eslint', { paths: [__filename] })` started to work. It was the version `2.3.3-git.20201112.f5ca1c57`.
Than I upgraded Yarn to the latest `2.4.0` and reinstalled dependencies again. And it still works. So I removed usage of PnP API as you asked